test/utils/test_convert.py

```diff
@@ -436,9 +436,9 @@ def test_to_cugraph(edge_weight, directed, relabel_nodes):
     cu_edge_index, cu_edge_weight = sort_edge_index(cu_edge_index,
                                                     cu_edge_weight)
 
-    assert torch.equal(edge_index, cu_edge_index)
```

Comment:
Can you clarify the fix for me? I would expect that both `edge_index` and `cu_edge_index` are on CPU?

Reply:
yeah im pretty sure cu_edge_index is coming out as cuda thats the issue. "As far as I can tell, the CI failures are because you are expecting a CPU tensor, but the cugraph edgelist is on the GPU.  That is the correct behavior from our end." i could also move them both back to cpu up to you, either one will fix

Reply:
I mean we are calling `to_pandas` that hold `numpy` values, so I would pretty amazed if they are not on the CPU.

Reply:
investigating further:
before sort cu_edge_index= tensor([[1, 0, 2, 1],
        [2, 1, 1, 0]], device='cuda:0')
so this implies [sorting the indices](https://github.com/pyg-team/pytorch_geometric/blob/fdb79725f303a36c22a548d918204041e95a133a/test/utils/test_convert.py#L477-L478) causes the dst nodes to be dropped for whatever reason.

[sorting code](https://github.com/pyg-team/pytorch_geometric/blob/fdb79725f303a36c22a548d918204041e95a133a/torch_geometric/utils/sort_edge_index.py#L71-L85)

Reply:
figured it out: https://github.com/pyg-team/pytorch_geometric/blob/fdb79725f303a36c22a548d918204041e95a133a/torch_geometric/utils/sort_edge_index.py#L80-L85

in these lines it should return the edge_index and then edge_attr but because the edge_attr is None then it only returns edge_index but we index into the output of the function w/  
cu_edge_index, cu_edge_weight = sort_edge_index(cu_edge_index, cu_edge_weight)
which splits src into `cu_edge_index` and dst into `cu_edge_weight`

soln:
cu_edge_index = sort_edge_index(cu_edge_index, cu_edge_weight).cpu()

Reply:
Got it. Guess we need to revisit the `sort_edge_index` interface. This is dangerous 😅